haystack/preview/components/fetchers/link_content.py

```diff
@@ -0,0 +1,225 @@
+        """
+        Fetches content from a URL specified by url parameter and converts it to a Document object.
+
+        param url: a URL link to fetch content from.
+
+        return: Document object
+        """
+        if not url:
+            raise ValueError("LinkContentFetcher run requires the `url` parameter")
+        return self.fetch(url=url)
+
+    def _register_content_handler(self, content_type: str, handler: Callable):
+        """
+        Register a new content handler for a specific content type.
+        If a handler for the given content type already exists, it will be overridden.
+
+        :param content_type: The content type for which the handler should be used.
+        :param handler: The handler function. This function should accept a requests.Response object parameter,
+        and return the extracted text, file-like object (or None).
+        """
+        if not callable(handler):
+            raise ValueError(f"handler must be a callable, but got {type(handler).__name__}")
+
+        params = inspect.signature(handler).parameters
+        if len(params) != 1 or list(params.keys()) != ["response"]:
+            raise ValueError(f"{content_type} handler must accept 'response: requests.Response' as a single parameter")
+
+        self.handlers[content_type] = handler
+
+    def _get_response(self, url: str, timeout: Optional[int] = None) -> requests.Response:
+        """
+        Fetches content from a URL. Returns a response object.
+        :param url: The URL to fetch content from.
+        :param timeout: The timeout in seconds.
+        :return: A response object.
+        """
+
+        @retry(
+            # we want to reraise the exception if we fail after the last self.retry_attempts
+            # then we can catch it in the outer try/except block, see below
+            reraise=True,
+            stop=stop_after_attempt(self.retry_attempts),
+            wait=wait_exponential(multiplier=1, min=2, max=10),
+            retry=(retry_if_exception_type((HTTPError, requests.RequestException))),
+            # This method is invoked only after failed requests (exception raised)
+            after=self._switch_user_agent,
+        )
+        def _request():
+            # we need a request copy because we modify the headers
+            headers = self._REQUEST_HEADERS.copy()
+            headers["User-Agent"] = self.user_agents[self.current_user_agent_idx]
+            r = requests.get(url, headers=headers, timeout=timeout or 3)
+            r.raise_for_status()
+            return r
+
+        try:
+            response = _request()
+        except Exception as e:
+            # catch all exceptions including HTTPError and RequestException
+            if self.raise_on_failure:
+                raise e
+            # if we don't raise on failure, log it, and return a response object
+            logger.warning("Couldn't retrieve content from %s", url)
+            response = requests.Response()
```

Comment:
If we're simplifying the `run` method as suggested, we can directly return `None` here.

```suggestion
            response = None
```